conans/test/functional/toolchains/google/test_bazel.py

```diff
@@ -0,0 +1,193 @@
+
+    cc_library(
+        name = "hello",
+        srcs = ["hello.cpp"],
+        hdrs = ["hello.h",],
+    )
+
+    cc_binary(
+        name = "main",
+        srcs = ["main.cpp"],
+        deps = [":hello"],
+    )
+""")
+
+    lib_h = gen_function_h(name="hello")
+    lib_cpp = gen_function_cpp(name="hello", msg="Hello", includes=["hello"])
+    main = gen_function_cpp(name="main", includes=["hello"], calls=["hello"])
+
+    workspace_file = textwrap.dedent("""
+    load("@//bazel-build/conandeps:dependencies.bzl", "load_conan_dependencies")
+    load_conan_dependencies()
+    """)
+
+    def setUp(self):
+        self.client = TestClient(path_with_spaces=False)  # bazel doesn't work with paths with spaces
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+            from conans.tools import save
+            import os
+            class Pkg(ConanFile):
+                settings = "build_type"
+                def package(self):
+                    save(os.path.join(self.package_folder, "include/hello.h"),
+                         '''#include <iostream>
+                         void hello(){std::cout<< "Hello: %s" <<std::endl;}'''
+                         % self.settings.build_type)
+            """)
+        self.client.save({"conanfile.py": conanfile})
+        self.client.run("create . hello/0.1@ -s build_type=Debug")
+        self.client.run("create . hello/0.1@ -s build_type=Release")
+
+        # Prepare the actual consumer package
+        self.client.save({"conanfile.py": self.conanfile,
+                          "WORKSPACE": self.workspace_file,
+                          "app/BUILD": self.buildfile,
+                          "app/main.cpp": self.main,
+                          "app/hello.cpp": self.lib_cpp,
+                          "app/hello.h": self.lib_h})
+
+    def _run_build(self):
+        build_directory = os.path.join(self.client.current_folder, "bazel-build").replace("\\", "/")
+        with self.client.chdir(build_directory):
+            self.client.run("install .. ")
+            install_out = self.client.out
+            self.client.run("build ..")
+        return install_out
+
+    def _modify_code(self):
+        lib_cpp = gen_function_cpp(name="hello", msg="HelloImproved", includes=["hello"])
+        self.client.save({"app/hello.cpp": lib_cpp})
+
+    def _incremental_build(self):
+        with self.client.chdir(self.client.current_folder):
+            self.client.run_command("bazel build --explain=output.txt //app:main")
+
+    def _run_app(self, bin_folder=False):
+        command_str = "bazel-bin/app/main.exe" if bin_folder else "bazel-bin/app/main"
+        if platform.system() == "Windows":
+            command_str = command_str.replace("/", "\\")
+
+        self.client.run_command(command_str)
+
+
+@pytest.mark.skipif(platform.system() != "Windows", reason="Only for windows")
+class WinTest(Base):
+    @parameterized.expand(["Debug"])
```

Comment:
Not very "expandable" right? LoL

Reply:
That's obviously temporal. The bazel generator can't handle build types yet, but it will 😄